articles/tutorials/building_2d_games/06_optimizing_texture_rendering/index.md

```diff
@@ -0,0 +1,503 @@
+- The `Width` and `Height` properties are get-only properties that are calculated based on the width and height of the `SourceRectangle` property, multiplied by the scale factor.  This automatically accounts for scaling, making it easier to perform calculations like collision detection or positioning without manually applying the scale factor each time.
+
+### Constructor
+Now that we have the properties defined, let's add a constructor:
+
+```cs
+/// <summary>
+/// Creates a new Sprite instance using the source texture and source rectangle provided.
+/// </summary>
+/// <param name="texture">The source texture of the sprite.</param>
+/// <param name="sourceRectangle">The source rectangle to use when rendering the sprite.</param>
+public Sprite(Texture2D texture, Rectangle sourceRectangle)
+{
+    Debug.Assert(texture is not null);
+    Debug.Assert(!texture.IsDisposed);
+
+    Texture = texture;
+    SourceRectangle = sourceRectangle;
+}
+```
+
+The constructor requires two parameters, a [**Texture2D**](xref:Microsoft.Xna.Framework.Graphics.Texture2D) and a [**Rectangle**](xref:Microsoft.Xna.Framework.Rectangle), representing the source texture (texture atlas) and the boundary within the atlas where the sprite is.  Before storing the references, checks are made to ensure:
+
+- The [**Texture2D**](xref:Microsoft.Xna.Framework.Graphics.Texture2D) given is not null
+- The [**Texture2D**](xref:Microsoft.Xna.Framework.Graphics.Texture2D) given was not previously disposed of
+- The [**Rectangle**](xref:Microsoft.Xna.Framework.Rectangle) given is a boundary that is contained within the [**Texture2D**](xref:Microsoft.Xna.Framework.Graphics.Texture2D).
+
+You might think that adding these checks are pointless, because when would you ever pass in a null or disposed texture, or provide a source rectangle that is out of bounds of the texture bounds. Of course you would never do this right?  Well, we're all human and sometimes we make mistakes.  It's always best to check yourself to be sure before you publish your game with bugs that could have been avoided.
+
+> [!TIP]  
+> Instead of throwing exceptions in the constructor when performing these checks,  `Debug.Assert` is used here.  This has a similar result as throwing an exception, except that the line of code is only ever executed when you run the code in a Debug build.  It asserts that the statement provided is true.  If the statement is false, then code execution will be paused at that line of code similar to if you add a breakpoint to debug.  This allows you to catch any issues while developing your game and running in a Debug build without needing to throw exceptions.  
```

Comment:
Include links to the MS source reference when referring to them.

Reply:
Link to namespaces added

Reply:
Links to MSDN article on Debug.Assert added